feat: add gcp_firestore_database table

pdecat opened the pull request:
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  uid,
  type,
  location
from
  gcp_firestore_database
where
  location = 'eur3';
+----------------------------------------+--------------------------------------+----------------+----------+
| name                                   | uid                                  | type           | location |
+----------------------------------------+--------------------------------------+----------------+----------+
| projects/myproject/databases/(default) | 2f6afdc0-33a3-4601-8356-c5c1589eafd4 | DATASTORE_MODE | eur3     |
+----------------------------------------+--------------------------------------+----------------+----------+
```
</details>

ParthaI reviewed (commented):
Hello @pdecat,

Thank you for the code changes for the new table addition — it looks great overall! I just had a few suggestions and clarifications I’d appreciate your input on:

- It seems the `FirestoreDatabaseService` function is missing in this PR. Could you please add it back?
- Based on the [API documentation](https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases#Database), it looks like some fields such as `cmekConfig`, `previousId`, and `sourceInfo` are not currently included in the table. Could you kindly double-check if we're missing any other fields returned by the API?
- Please ensure that error logs are added wherever they're missing. This will help with debugging in the future.
- The error logs should follow the same format used across other tables for consistency.
- Would it be possible to update the documentation for this table to align with the format we use for others? Specifically:
  - Include a clear title, description, and query explanation.
  - Provide queries in both PostgreSQL and SQLite formats.
  - If feasible, consider including a sample query that joins this table with others and/or demonstrates splitting JSON values into individual columns — similar to [this example](https://github.com/turbot/steampipe-plugin-gcp/blob/5f1ed34667697bc0633bf4932e1a7aed64298739/docs/tables/gcp_cloud_asset.md?plain=1#L45-L52).

Thanks again! Let me know if you have any questions.

pdecat commented:
Hi @ParthaI, sure sorry about all that, I rushed it a bit, and regarding the logging looked at the wrong tables as models (e.g. `table_gcp_dns_policy.go`).

I believe I've addressed most your comments except the advanced query example, PTAL :pray:

ParthaI reviewed (commented):
Hello @pdecat,

The changes look great to me overall — thank you for your contribution! I do have a couple of suggestions that might help polish things further. Could you please take another look?

- Would you mind reordering the table columns by moving the JSON-type columns toward the end, just before the standard Steampipe columns? This helps maintain consistency across our plugins.
- We generally avoid including `DEBUG`/`TRACE` log statements in the table functions, as they can create noise in the logs. Instead, we typically log only connection-related and API errors. For reference:
  - [Line 197](https://github.com/turbot/steampipe-plugin-gcp/pull/732/files#diff-d3235cf6fcf80a9dc8b85ab5432196a88d34afb7a51e6a3a4413dd30ffae41b8R197)
  - [Line 222](https://github.com/turbot/steampipe-plugin-gcp/pull/732/files#diff-d3235cf6fcf80a9dc8b85ab5432196a88d34afb7a51e6a3a4413dd30ffae41b8R222)

Would you be able to make these updates when you get a chance?

Thanks again! 🙌

pdecat commented:
Hi @ParthaI, done!

ParthaI commented on gcp/table_gcp_firestore_database.go:
```diff
@@ -0,0 +1,249 @@
+
+	// List databases
+	parent := "projects/" + project
+	resp, err := service.Projects.Databases.List(parent).ShowDeleted(true).Do()
```
Should we always hardcode the value `ShowDeleted(true)`?

It might be more flexible if we give users the option to include or exclude deleted databases based on their needs. If someone prefers not to include deleted databases in the response, we should allow them to filter those out.

Do we know what the default behavior of the API is? Does it return deleted databases by default, or are they excluded unless explicitly requested?

Ideally, when running a query like:

```sql
select * from gcp_firestore_database where show_shadow
```

the table should return **all** databases (including deleted ones), only when the `show_shadow` qualifier is set.

Would it be possible to support an optional qualifier named `show_shadow`, and use a transform function like `transform.FromQual("show_shadow")` to populate the column?

For reference, a similar pattern is used in the `gcp_logging_log_entry` table with the `filter` column.

Thanks in advance for considering this!

pdecat replied:
Added:
```
> select show_deleted, create_time, delete_time from gcp_firestore_database
+--------------+-------------+-------------+
| show_deleted | create_time | delete_time |
+--------------+-------------+-------------+
+--------------+-------------+-------------+
> select show_deleted, create_time, delete_time from gcp_firestore_database where show_deleted
+--------------+---------------------------+---------------------------+
| show_deleted | create_time               | delete_time               |
+--------------+---------------------------+---------------------------+
| true         | 2017-10-30T12:08:02+01:00 | 2025-03-27T17:06:24+01:00 |
| true         | 2025-03-27T16:52:05+01:00 | 2025-03-27T17:06:00+01:00 |
+--------------+---------------------------+---------------------------+
> select show_deleted, create_time, delete_time from gcp_firestore_database where not show_deleted
+--------------+-------------+-------------+
| show_deleted | create_time | delete_time |
+--------------+-------------+-------------+
+--------------+-------------+-------------+

```

ParthaI reviewed (commented):
Hello @pdecat, I’ve added a comment regarding the `ShowDeleted` input parameter. When you get a chance, could you please take a look? Thanks!

ParthaI commented on gcp/table_gcp_firestore_database.go:
```diff
@@ -0,0 +1,266 @@
+			{
+				Name:        "show_deleted",
+				Type:        proto.ColumnType_BOOL,
+				Description: "Should deleted databases be shown?",
```
Just a small suggestion — could we update the description to something like:  
`"Set to true to include deleted databases in the query results. By default, deleted databases are excluded."`  
I think this wording would be a bit clearer for users.

ParthaI reviewed (approved):
Thanks @pdecat, approved!
